src/posix/platform/radio_spinel.cpp

```diff
@@ -452,6 +488,160 @@ void RadioSpinel::HandleResponse(const uint8_t *aBuffer, uint16_t aLength)
     LogIfFail("Error processing response", error);
 }
 
+otError RadioSpinel::ThreadDatasetHandler(const uint8_t *aBuffer, uint16_t aLength)
+{
+    otError              error = OT_ERROR_NONE;
+    otOperationalDataset opDataset;
+    Ncp::SpinelDecoder   decoder;
+    MeshCoP::Dataset     dataset((mWaitingKey == SPINEL_PROP_THREAD_ACTIVE_DATASET) ? MeshCoP::Tlv::kActiveTimestamp
+                                                                                : MeshCoP::Tlv::kPendingTimestamp);
+
+    memset(&opDataset, 0, sizeof(otOperationalDataset));
+    decoder.Init(aBuffer, aLength);
+
+    while (!decoder.IsAllReadInStruct())
+    {
+        unsigned int propKey;
+
+        SuccessOrExit(error = decoder.OpenStruct());
+        SuccessOrExit(error = decoder.ReadUintPacked(propKey));
+
+        switch (static_cast<spinel_prop_key_t>(propKey))
+        {
+        case SPINEL_PROP_NET_MASTER_KEY:
+        {
+            const uint8_t *key;
+            uint16_t       len;
+
+            SuccessOrExit(error = decoder.ReadData(key, len));
+            VerifyOrExit(len == OT_MASTER_KEY_SIZE, error = OT_ERROR_INVALID_ARGS);
+            memcpy(opDataset.mMasterKey.m8, key, len);
+            opDataset.mComponents.mIsMasterKeyPresent = true;
+            break;
+        }
+
+        case SPINEL_PROP_NET_NETWORK_NAME:
+        {
+            const char *name;
+            size_t      len;
+
+            SuccessOrExit(error = decoder.ReadUtf8(name));
+            len = strnlen(name, OT_NETWORK_NAME_MAX_SIZE + 1);
+            memcpy(opDataset.mNetworkName.m8, name, len + 1);
```

Comment:
What happens when `len` is `OT_NETWORK_NAME_MAX_SIZE + 1`?

Reply:
You are right. It will probably copy a name string without tailing \0. I think the network name should not exceed OT_NETWORK_NAME_MAX_SIZE, otherwise, an assert should happen here.